packages/dds/tree/src/simple-tree/core/getOrCreateNode.ts

```diff
@@ -0,0 +1,30 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import type { TreeValue } from "../../core/index.js";
+import { isMapTreeNode, type FlexTreeNode } from "../../feature-libraries/index.js";
+import { fail } from "../../util/index.js";
+import { type InnerNode, mapTreeNodeToProxy, proxySlot } from "./treeNodeKernel.js";
+import { getSimpleNodeSchemaFromInnerNode } from "./schemaCaching.js";
+import type { TreeNode, InternalTreeNode } from "./types.js";
+
+export function getOrCreateNodeFromInnerNode(flexNode: InnerNode): TreeNode | TreeValue {
```

Comment:
Docs?

Reply:
Note that this isn't new code, its moved and renamed from the misleadingly named `getOrCreateNodeFromFlexTreeNode`, which didn't have any docs. I'll still add docs though since I had to figure out what this does anyway.